File: modules/sqlite-jdbc/src/main/java/io/roastedroot/sqlite4j/util/StringUtils.java
package io.roastedroot.sqlite4j.util;

import java.util.List;
import org.jspecify.annotations.NonNull;

public class StringUtils {
    public static String join(@NonNull List<String> list, @NonNull String separator) {
        StringBuilder sb = new StringBuilder();
        boolean first = true;
        for (String item : list) {
            if (first) first = false;
            else sb.append(separator);

            sb.append(item);
        }
        return sb.toString();
    }
}

File: modules/sqlite-jdbc/src/main/java/io/roastedroot/sqlite4j/util/StringUtils.kt
package io.roastedroot.sqlite4j.util

object StringUtils {
    @JvmStatic
    fun join(list: List<String>, separator: String): String {
        val sb = StringBuilder()
        var first = true
        for (item in list) {
            if (first) first = false
            else sb.append(separator)

            sb.append(item)
        }
        return sb.toString()
    }
}
